india_compliance/gst_india/report/gst_job_work_stock_movement/gst_job_work_stock_movement.js

```diff
@@ -57,4 +57,104 @@ frappe.query_reports["GST Job Work Stock Movement"] = {
             reqd: 1,
         },
     ],
+    onload: function (query_report) {
+        query_report.page.add_inner_button(__("Export JSON"), function () {
+            this.dialog = new frappe.ui.Dialog({
+                title: __("Export JSON"),
+                fields: [
+                    {
+                        fieldname: "year",
+                        label: __("Year"),
+                        fieldtype: "Select",
+                        options: get_options_for_year(),
+                        onchange: () => {
+                            parseInt(this.dialog.get_value("year"), 10) <= 2021 &&
+                                this.dialog.set_value("return_type", ""); // Annual return not available before 2022
+
+                            this.dialog.fields_dict.period.df.options =
+                                get_options_for_month(this.dialog.get_value("year"));
+                            this.dialog.refresh();
+                        },
+                        reqd: 1,
+                    },
+                    {
+                        fieldname: "return_type",
+                        label: __("Return Type"),
+                        fieldtype: "Select",
+                        options: ["", "Annual", "Half Yearly"],
+                        depends_on: "eval: parseInt(doc.year) > 2021",
+                    },
+                    {
+                        fieldname: "period",
+                        label: __("Return Filing Period"),
+                        fieldtype: "Select",
+                        depends_on: "eval: doc.return_type !== 'Annual'",
+                    },
```

Comment:
Why not guess the period => validate it => use it for data export. Dialog is not a good ux.